src/videojs.ads.js

```diff
@@ -337,43 +338,119 @@ var
       };
 
       player.on(videoEvents, function redispatch(event) {
-        if (player.ads.state === 'ad-playback') {
+
+        // We do a quick play/pause before we check for prerolls. This creates a "playing"
+        // event. This conditional block prefixes that event so it's "adplaying" if it
+        // happens while we're in the "preroll?" state. Not every browser is in the
+        // "preroll?" state for this event, so the following browsers come through here:
+        //  * iPad
+        //  * iPhone
+        //  * Android
+        //  * Safari
+        // This is too soon to check videoElementRecycled because there is no snapshot
+        // yet. We rely on the coincidence that all browsers for which
+        // videoElementRecycled would be true also happen to send their initial playing
+        // event during "preroll?"
+        if (event.type === 'playing' && player.ads.state === 'preroll?') {
+          triggerEvent('ad', event);
+
+        // Here we send "adplaying" for browsers that send their initial "playing" event
+        // (caused by the the initial play/pause) during the "ad-playback" state.
+        // The following browsers come through here:
+        // * Chrome
+        // * IE11
+        // If the ad plays in the content tech (aka videoElementRecycled) there will be
+        // another playing event when the ad starts. We check videoElementRecycled to
+        // avoid a second adplaying event. Thankfully, at this point a snapshot exists
+        // so we can safely check videoElementRecycled.
+        } else if (event.type === 'playing' &&
+            player.ads.state === 'ad-playback' &&
+            !player.ads.videoElementRecycled()) {
+          triggerEvent('ad', event);
+
+        // When an ad is playing in content tech, we would normally prefix
+        // "playing" with "ad" to send "adplaying". However, when we did a play/pause
+        // before the preroll, we already sent "adplaying". This condition prevents us
+        // from sending another.
+        } else if (event.type === 'playing' &&
+            player.ads.state === 'ad-playback' &&
+            player.ads.videoElementRecycled()) {
+
+          // Triggering an event prevents the unprefixed one from firing.
+          // "adcontentplaying" is only seen in this very specific condition.
+          triggerEvent('adcontent', event);
+          return;
+
+        // When ad is playing in content tech, prefix everything with "ad".
+        // This block catches many events such as emptied, play, timeupdate, and ended.
+        } else if (player.ads.state === 'ad-playback') {
```

Comment:
Yeah I already know of one place where the change to "playing" would break something else. This will be tricky